runtime/README.md

```diff
@@ -1226,129 +1237,239 @@ Command Code: `0x434D_4B54` ("CMKT")
 | **Name**        | **Type** | **Description** |
 | --------------- | -------- | --------------- |
 | tag size        | u32      |                 |
+| tag             | u8[16]   | padded with 0s  |
 | last length     | u32      |                 |
 | last iv         | u8[16]   |                 |
 | last GHASH      | u8[16]   |                 |
 | last ciphertext | u8[16]   |                 |
 
+### CM_AES_GCM_DECRYPT_UPDATE
+
+This continues an AES computation started by `CM_AES_GCM_DECRYPT_INIT` or from another `CM_AES_GCM_DECRYPT_UPDATE`.
+
+The context MUST be passed in from `CM_AES_GCM_DECRYPT_INIT` or `CM_AES_GCM_DECRYPT_UPDATE`.
+
+Command Code: `0x434D_4455` ("CMDU")
+
+*Table: `CM_AES_GCM_DECRYPT_UPDATE` input arguments*
+| **Name**        | **Type** | **Description**     |
+| --------------- | -------- | ------------------- |
+| chksum          | u32      |                     |
+| context size    | u32      | Size of the context |
+| context         | u8[...]  |                     |
+| ciphertext size | u32      | MUST be non-zero    |
+| ciphertext      | u8[...]  | Data to decrypt     |
+
+*Table: `CM_AES_GCM_DECRYPT_UPDATE` output arguments*
+| **Name**       | **Type** | **Description**           |
+| -------------- | -------- | ------------------------- |
+| chksum         | u32      |                           |
+| fips_status    | u32      | FIPS approved or an error |
+| context size   | u32      |                           |
+| context        | u8[...]  |                           |
+| plaintext size | u32      | MAY be 0                  |
+| plaintext      | u8[...]  |                           |
+
+### CM_AES_GCM_DECRYPT_FINAL
+
+This finalizes the computation of the AES GCM decryption and produces the final ciphertext.
+
+The context MUST be passed in from `CM_AES_GCM_DECRYPT_INIT` or `CM_AES_GCM_DECRYPT_UPDATE`.
+
+Command Code: `0x434D_4446` ("CMDF")
+
+*Table: `CM_AES_GCM_DECRYPT_FINAL` input arguments*
+| **Name**        | **Type** | **Description**                   |
+| --------------- | -------- | --------------------------------- |
+| chksum          | u32      |                                   |
+| flags           | u32      | Bit 0 = this is the final message |
+| context size    | u32      | Size of the context               |
+| context         | u8[...]  |                                   |
+| ciphertext size | u32      | MAY be 0                          |
+| ciphertext      | u8[...]  | Data to decrypt                   |
+
+*Table: `CM_AES_GCM_DECRYPT_FINAL` output arguments*
+| **Name**       | **Type** | **Description**                      |
+| -------------- | -------- | ------------------------------------ |
+| chksum         | u32      |                                      |
+| fips_status    | u32      | FIPS approved or an error            |
+| tag verified   | u32      | 1 if tags matched, 0 if they did not |
+| tag            | u8[16]   | Computed tag                         |
```

Comment:
Do we need to return the tag if it was given in the init stage?

Reply:
If the tag did not match, it might be useful for debugging to know what the hardware thought the tag should be.